Build errors

I tried to build the source code on a Raspberry Pi 3 B+ and got the following errors running make:

/usr/bin/ld: CMakeFiles/qcx_cat_ft8.dir/src/qcx_cat_fsk/audio_decoder.cc.o: in function `qcx_cat_fsk::AudioDecoder::Decode()':
audio_decoder.cc:(.text+0x4f4): undefined reference to `boost::log::v2s_mt_posix::trivial::logger::get()'
/usr/bin/ld: audio_decoder.cc:(.text+0x558): undefined reference to `boost::log::v2s_mt_posix::trivial::logger::get()'
/usr/bin/ld: audio_decoder.cc:(.text+0x780): undefined reference to `boost::log::v2s_mt_posix::trivial::logger::get()'
/usr/bin/ld: audio_decoder.cc:(.text+0x7e4): undefined reference to `boost::log::v2s_mt_posix::trivial::logger::get()'
/usr/bin/ld: audio_decoder.cc:(.text+0x930): undefined reference to `boost::log::v2s_mt_posix::trivial::logger::get()'
...

Can you help me solve this problem?

Thanks,
Jim N7IHQ

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,9 @@ cmake_minimum_required(VERSION 3.0)
 project(qcx_cat_fsk)
 
 find_package(PulseAudio REQUIRED)
-find_package(Boost REQUIRED COMPONENTS log log_setup)
+find_package(Boost COMPONENTS log log_setup REQUIRED)
+
+add_definitions(-DBOOST_LOG_DYN_LINK)
 
 include_directories(
     ${CMAKE_SOURCE_DIR}/include
@@ -18,5 +20,6 @@ add_executable(qcx_cat_ft8
 
 set_property(TARGET qcx_cat_ft8 PROPERTY CXX_STANDARD 17)
 
+
 target_link_libraries(qcx_cat_ft8
-    ${PULSEAUDIO_LIBRARY} pulse-simple ${Boost_LIBRARIES} pthread)
+    ${PULSEAUDIO_LIBRARY} pulse-simple ${Boost_LOG_LIBRARY} ${Boost_LOG_SETUP_LIBRARY} pthread)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,4 +22,4 @@ set_property(TARGET qcx_cat_ft8 PROPERTY CXX_STANDARD 17)
 
 
 target_link_libraries(qcx_cat_ft8
-    ${PULSEAUDIO_LIBRARY} pulse-simple ${Boost_LOG_LIBRARY} ${Boost_LOG_SETUP_LIBRARY} pthread)
+    ${PULSEAUDIO_LIBRARY} pulse-simple ${Boost_LIBRARIES} pthread)

--- a/src/main.cc
+++ b/src/main.cc
@@ -15,6 +15,8 @@
  * along with qcx_cat_fsk.  If not, see <https://www.gnu.org/licenses/>.
  */
 
+#define BOOST_LOG_DYN_LINK 1
+
 #include <boost/log/core.hpp>
 #include <boost/log/expressions.hpp>
 #include <boost/log/trivial.hpp>

--- a/src/main.cc
+++ b/src/main.cc
@@ -15,8 +15,6 @@
  * along with qcx_cat_fsk.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-#define BOOST_LOG_DYN_LINK 1
-
 #include <boost/log/core.hpp>
 #include <boost/log/expressions.hpp>
 #include <boost/log/trivial.hpp>

--- a/src/qcx_cat_fsk/audio_decoder.cc
+++ b/src/qcx_cat_fsk/audio_decoder.cc
@@ -37,7 +37,7 @@ AudioDecoder::AudioDecoder(const std::shared_ptr<CatInterface> &cat_interface,
   /* Create the recording stream */
   int error;
   if (!(this->audio_stream_ =
-            pa_simple_new(NULL, "qcx_cat_fsk", PA_STREAM_RECORD, NULL, "record",
+            pa_simple_new(NULL, "qcx_cat_ft8", PA_STREAM_RECORD, NULL, "record",
                           &ss, NULL, NULL, &error))) {
     throw std::runtime_error(pa_strerror(error));
   }

